Better error message when posterior is empty

Right now, the error is:

```
AssertionError: Expected marginal to be normalized.
    at Object.makeMarginalERP (/Users/long/webppl/src/erp.js:608:12)
    at Distribution.toERP (/Users/long/webppl/src/aggregation.js:57:14)
    at Enumerate.exit (/Users/long/webppl/src/inference/enumerate.js:135:47)
```

which is likely inscrutable for most users.

--- a/src/aggregation.js
+++ b/src/aggregation.js
@@ -35,6 +35,10 @@ var Distribution = function() {
   this.dist = {};
 };
 
+Object.defineProperties(Distribution.prototype, {
+  size: { get: function() { return _.size(this.dist); } }
+});
+
 Distribution.prototype.add = function(value, score) {
   if (score === -Infinity) {
     return;

--- a/src/inference/enumerate.js
+++ b/src/inference/enumerate.js
@@ -129,6 +129,9 @@ module.exports = function(env) {
     if (this.queue.size() > 0 && (this.numCompletedExecutions < this.maxExecutions)) {
       return this.nextInQueue();
     } else {
+      if (this.marginal.size === 0) {
+        throw 'All paths explored by Enumerate have probability zero.';
+      }
       // Reinstate previous coroutine:
       env.coroutine = this.coroutine;
       // Return from enumeration by calling original continuation with original store:

